cpp/include/raft/matrix/slice.cuh

```diff
@@ -45,17 +46,18 @@ struct slice_coordinates {
  * @tparam m_t type of matrix elements
  * @tparam idx_t integer type used for indexing
  * @param[in] handle: raft handle
- * @param[in] in: input matrix (column-major)
- * @param[out] out: output matrix (column-major)
+ * @param[in] in: input matrix
+ * @param[out] out: output matrix
  * @param[in] coords: coordinates of the wanted slice
  * example: Slice the 2nd and 3rd columns of a 4x3 matrix: slice(handle, in, out, {0, 1, 4, 3});
  */
-template <typename m_t, typename idx_t>
+template <typename m_t, typename idx_t, typename layout_t>
 void slice(raft::resources const& handle,
-           raft::device_matrix_view<const m_t, idx_t, col_major> in,
-           raft::device_matrix_view<m_t, idx_t, col_major> out,
+           raft::device_matrix_view<const m_t, idx_t, layout_t> in,
+           raft::device_matrix_view<m_t, idx_t, layout_t> out,
            slice_coordinates<idx_t> coords)
 {
+  RAFT_EXPECTS(raft::is_row_or_column_major(in), "Unsupported matrix layout");
```

Comment:
In addition to the assertion, can we provide a little hint to the user to help them fix this? Maybe something like "Matrix layout must be row- or column-major".